.github/workflows/publish-branch.yaml

```diff
@@ -0,0 +1,77 @@
+name: Publish Branch
+
+on:
+    push:
+        branches:
+            - main
+            - dev
+
+jobs:
+    release_candidate:
+        runs-on: ubuntu-latest
+        env:
+            NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+            NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+            BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  ref: ${{ github.head_ref }}
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: "16.x"
+                  registry-url: "https://registry.npmjs.org"
+                  cache: "yarn"
+
+            - name: Install dev dependencies
+              run: yarn install
+
+            - name: Setup npm credentials file
+              run: echo "//registry.npmjs.org/:_authToken=$NODE_AUTH_TOKEN" >> .npmrc
+
+            # - name: Setup git credentials
+            #   run: |
+            #       git config --global user.name 'Auto Release Bot'
+            #       git config --global user.email 'auto-release-bot@users.noreply.github.com'
+
+            - name: Get current package.json version
+              run: echo "PACKAGE_VERSION=$(npm pkg get version | tr -d '"')" >> $GITHUB_ENV
+
+            # get the sort SHA and add it into the environment variables
+            - name: Get short SHA
+              run: echo "SHORT_SHA=${GITHUB_SHA::7}" >> $GITHUB_ENV
+
+            # get the package name so the workflow can be reusable between projects
+            # - name: Get current package.json name
+            #   run: echo "PACKAGE_NAME=$(npm pkg get name | tr -d '"')" >> $GITHUB_ENV
+
+            - name: Setup Branch Release Candidate Version
+              run: echo "NEW_VERSION=$PACKAGE_VERSION-$BRANCH_NAME.$SHORT_SHA" >> $GITHUB_ENV
+
+            - name: Update Package.json Version with Release Candidate Version
+              run: node ./update-branch-version.js --version $NEW_VERSION
```

Comment:
I tried a lot to discard the need for this JS file, I did the entire logic on the workflow, but when triggered we received the following error:

![image](https://github.com/pixijs/pixijs/assets/9615850/fc398ce7-0281-4eb4-a8de-0996b83cbcbf)

I tried 16 possible changes to debug and fix it and it doesn't work as expected, but adding it to a JS file and running it through the node command worked as expected

Reply:
This seems very complicated. Why aren't you just doing `run: npm version $NEW_VERSION`?

Reply:
Because it didn't work, I moved the entire logic to the workflow (because is simple now), but the Action was not able to trigger the npm version command on this format, not even hard coded